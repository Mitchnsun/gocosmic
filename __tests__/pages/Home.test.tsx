import type { AnchorHTMLAttributes, ReactNode } from 'react';
import React from 'react';
import { vi } from 'vitest';

import Home from '@/app/[locale]/page';

import { render } from '../test-utils';

type MockLinkProps = { href: string; children: ReactNode } & AnchorHTMLAttributes<HTMLAnchorElement>;

vi.mock('@/i18n/navigation', () => ({
  Link: ({ children, href, ...props }: MockLinkProps) => (
    <a href={href} {...props}>
      {children}
    </a>
  ),
}));

describe('Home Page', () => {
  it('should render the home page correctly', () => {
    const { container } = render(<Home />);

    expect(container).toMatchSnapshot();
  });
});
