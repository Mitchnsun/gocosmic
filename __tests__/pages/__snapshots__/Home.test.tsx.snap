// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Home Page > should render the home page correctly 1`] = `
<div>
  <div
    class="text-ghost relative p-4"
  >
    <main
      class="m-auto flex max-w-7xl flex-col items-center py-16 md:px-4"
    >
      <div
        class="mb-4 flex flex-col items-center gap-3 md:flex-row"
      >
        <svg
          aria-hidden="true"
          class="h-10 w-10 text-blue-400"
          data-slot="icon"
          fill="currentColor"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M9.315 7.584C12.195 3.883 16.695 1.5 21.75 1.5a.75.75 0 0 1 .75.75c0 5.056-2.383 9.555-6.084 12.436A6.75 6.75 0 0 1 9.75 22.5a.75.75 0 0 1-.75-.75v-4.131A15.838 15.838 0 0 1 6.382 15H2.25a.75.75 0 0 1-.75-.75 6.75 6.75 0 0 1 7.815-6.666ZM15 6.75a2.25 2.25 0 1 0 0 4.5 2.25 2.25 0 0 0 0-4.5Z"
            fill-rule="evenodd"
          />
          <path
            d="M5.26 17.242a.75.75 0 1 0-.897-1.203 5.243 5.243 0 0 0-2.05 5.022.75.75 0 0 0 .625.627 5.243 5.243 0 0 0 5.022-2.051.75.75 0 1 0-1.202-.897 3.744 3.744 0 0 1-3.008 1.51c0-1.23.592-2.323 1.51-3.008Z"
          />
        </svg>
        <h2
          class="text-center text-2xl font-extrabold sm:text-4xl"
        >
          We build stellar applications that shine
        </h2>
        <svg
          aria-hidden="true"
          class="h-8 w-8 text-yellow-400"
          data-slot="icon"
          fill="currentColor"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M9 4.5a.75.75 0 0 1 .721.544l.813 2.846a3.75 3.75 0 0 0 2.576 2.576l2.846.813a.75.75 0 0 1 0 1.442l-2.846.813a3.75 3.75 0 0 0-2.576 2.576l-.813 2.846a.75.75 0 0 1-1.442 0l-.813-2.846a3.75 3.75 0 0 0-2.576-2.576l-2.846-.813a.75.75 0 0 1 0-1.442l2.846-.813A3.75 3.75 0 0 0 7.466 7.89l.813-2.846A.75.75 0 0 1 9 4.5ZM18 1.5a.75.75 0 0 1 .728.568l.258 1.036c.236.94.97 1.674 1.91 1.91l1.036.258a.75.75 0 0 1 0 1.456l-1.036.258c-.94.236-1.674.97-1.91 1.91l-.258 1.036a.75.75 0 0 1-1.456 0l-.258-1.036a2.625 2.625 0 0 0-1.91-1.91l-1.036-.258a.75.75 0 0 1 0-1.456l1.036-.258a2.625 2.625 0 0 0 1.91-1.91l.258-1.036A.75.75 0 0 1 18 1.5ZM16.5 15a.75.75 0 0 1 .712.513l.394 1.183c.15.447.5.799.948.948l1.183.395a.75.75 0 0 1 0 1.422l-1.183.395c-.447.15-.799.5-.948.948l-.395 1.183a.75.75 0 0 1-1.422 0l-.395-1.183a1.5 1.5 0 0 0-.948-.948l-1.183-.395a.75.75 0 0 1 0-1.422l1.183-.395c.447-.15.799-.5.948-.948l.395-1.183A.75.75 0 0 1 16.5 15Z"
            fill-rule="evenodd"
          />
        </svg>
      </div>
      <p
        class="mb-8 text-center"
      >
        Let us launch your ideas into the stratosphere with mystical design and cosmic functionality
      </p>
      <a
        aria-label="Start your journey"
        class="inline-flex items-center justify-center cursor-pointer rounded-full transition-colors disabled:pointer-events-none disabled:opacity-50 bg-royal text-neutral-50 hover:bg-royal/90 font-normal text-base px-6 py-2 gap-2"
        href="/journey"
      >
        <svg
          aria-hidden="true"
          class="h-5 w-5"
          data-slot="icon"
          fill="currentColor"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M10.788 3.21c.448-1.077 1.976-1.077 2.424 0l2.082 5.006 5.404.434c1.164.093 1.636 1.545.749 2.305l-4.117 3.527 1.257 5.273c.271 1.136-.964 2.033-1.96 1.425L12 18.354 7.373 21.18c-.996.608-2.231-.29-1.96-1.425l1.257-5.273-4.117-3.527c-.887-.76-.415-2.212.749-2.305l5.404-.434 2.082-5.005Z"
            fill-rule="evenodd"
          />
        </svg>
        Start your journey
      </a>
    </main>
    <br />
    <h3
      class="text-center text-2xl font-bold sm:text-3xl"
      id="services"
    >
      Our Cosmic Services
    </h3>
    <div
      class="m-auto grid max-w-7xl grid-cols-1 items-stretch justify-center gap-8 py-8 md:grid-cols-2 xl:grid-cols-4"
    >
      <div
        class="flex flex-col items-center justify-around rounded-lg bg-gray-800 px-2 py-4"
      >
        <svg
          aria-hidden="true"
          class="text-jungle h-6 w-6"
          data-slot="icon"
          fill="currentColor"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M14.447 3.026a.75.75 0 0 1 .527.921l-4.5 16.5a.75.75 0 0 1-1.448-.394l4.5-16.5a.75.75 0 0 1 .921-.527ZM16.72 6.22a.75.75 0 0 1 1.06 0l5.25 5.25a.75.75 0 0 1 0 1.06l-5.25 5.25a.75.75 0 1 1-1.06-1.06L21.44 12l-4.72-4.72a.75.75 0 0 1 0-1.06Zm-9.44 0a.75.75 0 0 1 0 1.06L2.56 12l4.72 4.72a.75.75 0 0 1-1.06 1.06L.97 12.53a.75.75 0 0 1 0-1.06l5.25-5.25a.75.75 0 0 1 1.06 0Z"
            fill-rule="evenodd"
          />
        </svg>
        <h4
          class="my-2 text-center text-xl font-semibold sm:text-2xl"
        >
          Stellar Development
        </h4>
        <p
          class="text-center font-light text-gray-400"
        >
          Building robust and scalable mobile applications for both iOS and Android platforms.
        </p>
      </div>
      <div
        class="flex flex-col items-center justify-around rounded-lg bg-gray-800 px-2 py-4"
      >
        <svg
          aria-hidden="true"
          class="text-royal h-6 w-6"
          data-slot="icon"
          fill="currentColor"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M9 4.5a.75.75 0 0 1 .721.544l.813 2.846a3.75 3.75 0 0 0 2.576 2.576l2.846.813a.75.75 0 0 1 0 1.442l-2.846.813a3.75 3.75 0 0 0-2.576 2.576l-.813 2.846a.75.75 0 0 1-1.442 0l-.813-2.846a3.75 3.75 0 0 0-2.576-2.576l-2.846-.813a.75.75 0 0 1 0-1.442l2.846-.813A3.75 3.75 0 0 0 7.466 7.89l.813-2.846A.75.75 0 0 1 9 4.5ZM18 1.5a.75.75 0 0 1 .728.568l.258 1.036c.236.94.97 1.674 1.91 1.91l1.036.258a.75.75 0 0 1 0 1.456l-1.036.258c-.94.236-1.674.97-1.91 1.91l-.258 1.036a.75.75 0 0 1-1.456 0l-.258-1.036a2.625 2.625 0 0 0-1.91-1.91l-1.036-.258a.75.75 0 0 1 0-1.456l1.036-.258a2.625 2.625 0 0 0 1.91-1.91l.258-1.036A.75.75 0 0 1 18 1.5ZM16.5 15a.75.75 0 0 1 .712.513l.394 1.183c.15.447.5.799.948.948l1.183.395a.75.75 0 0 1 0 1.422l-1.183.395c-.447.15-.799.5-.948.948l-.395 1.183a.75.75 0 0 1-1.422 0l-.395-1.183a1.5 1.5 0 0 0-.948-.948l-1.183-.395a.75.75 0 0 1 0-1.422l1.183-.395c.447-.15.799-.5.948-.948l.395-1.183A.75.75 0 0 1 16.5 15Z"
            fill-rule="evenodd"
          />
        </svg>
        <h4
          class="my-2 text-center text-xl font-semibold sm:text-2xl"
        >
          Mystical UI/UX Design
        </h4>
        <p
          class="text-center font-light text-gray-400"
        >
          Crafting intuitive and enchanting user experiences that captivate and delight.
        </p>
      </div>
      <div
        class="flex flex-col items-center justify-around rounded-lg bg-gray-800 px-2 py-4"
      >
        <svg
          aria-hidden="true"
          class="h-6 w-6 text-yellow-400"
          data-slot="icon"
          fill="currentColor"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M11.25 5.337c0-.355-.186-.676-.401-.959a1.647 1.647 0 0 1-.349-1.003c0-1.036 1.007-1.875 2.25-1.875S15 2.34 15 3.375c0 .369-.128.713-.349 1.003-.215.283-.401.604-.401.959 0 .332.278.598.61.578 1.91-.114 3.79-.342 5.632-.676a.75.75 0 0 1 .878.645 49.17 49.17 0 0 1 .376 5.452.657.657 0 0 1-.66.664c-.354 0-.675-.186-.958-.401a1.647 1.647 0 0 0-1.003-.349c-1.035 0-1.875 1.007-1.875 2.25s.84 2.25 1.875 2.25c.369 0 .713-.128 1.003-.349.283-.215.604-.401.959-.401.31 0 .557.262.534.571a48.774 48.774 0 0 1-.595 4.845.75.75 0 0 1-.61.61c-1.82.317-3.673.533-5.555.642a.58.58 0 0 1-.611-.581c0-.355.186-.676.401-.959.221-.29.349-.634.349-1.003 0-1.035-1.007-1.875-2.25-1.875s-2.25.84-2.25 1.875c0 .369.128.713.349 1.003.215.283.401.604.401.959a.641.641 0 0 1-.658.643 49.118 49.118 0 0 1-4.708-.36.75.75 0 0 1-.645-.878c.293-1.614.504-3.257.629-4.924A.53.53 0 0 0 5.337 15c-.355 0-.676.186-.959.401-.29.221-.634.349-1.003.349-1.036 0-1.875-1.007-1.875-2.25s.84-2.25 1.875-2.25c.369 0 .713.128 1.003.349.283.215.604.401.959.401a.656.656 0 0 0 .659-.663 47.703 47.703 0 0 0-.31-4.82.75.75 0 0 1 .83-.832c1.343.155 2.703.254 4.077.294a.64.64 0 0 0 .657-.642Z"
          />
        </svg>
        <h4
          class="my-2 text-center text-xl font-semibold sm:text-2xl"
        >
          AI Powered
        </h4>
        <p
          class="text-center font-light text-gray-400"
        >
          Integrating advanced AI features to make your applications smarter, more adaptive, and truly next-gen.
        </p>
      </div>
      <div
        class="flex flex-col items-center justify-around rounded-lg bg-gray-800 px-2 py-4"
      >
        <svg
          aria-hidden="true"
          class="text-aerospace h-6 w-6"
          data-slot="icon"
          fill="currentColor"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M9.315 7.584C12.195 3.883 16.695 1.5 21.75 1.5a.75.75 0 0 1 .75.75c0 5.056-2.383 9.555-6.084 12.436A6.75 6.75 0 0 1 9.75 22.5a.75.75 0 0 1-.75-.75v-4.131A15.838 15.838 0 0 1 6.382 15H2.25a.75.75 0 0 1-.75-.75 6.75 6.75 0 0 1 7.815-6.666ZM15 6.75a2.25 2.25 0 1 0 0 4.5 2.25 2.25 0 0 0 0-4.5Z"
            fill-rule="evenodd"
          />
          <path
            d="M5.26 17.242a.75.75 0 1 0-.897-1.203 5.243 5.243 0 0 0-2.05 5.022.75.75 0 0 0 .625.627 5.243 5.243 0 0 0 5.022-2.051.75.75 0 1 0-1.202-.897 3.744 3.744 0 0 1-3.008 1.51c0-1.23.592-2.323 1.51-3.008Z"
          />
        </svg>
        <h4
          class="my-2 text-center text-xl font-semibold sm:text-2xl"
        >
          Cosmic Launch
        </h4>
        <p
          class="text-center font-light text-gray-400"
        >
          Guiding your app through the launch process and beyond, ensuring a successful liftoff.
        </p>
      </div>
    </div>
    <div
      class="m-auto flex max-w-7xl flex-col items-center gap-2 rounded-lg bg-gray-800 px-4 py-12"
    >
      <h3
        class="py-2 text-center text-xl font-bold sm:text-3xl"
      >
        Ready to Go Cosmic?
      </h3>
      <p
        class="text-center font-light text-gray-400"
      >
        Let's discuss how we can bring your mobile or web app vision to life, creating something truly out of this world.
      </p>
      <a
        aria-label="View our offers"
        class="inline-flex items-center justify-center cursor-pointer rounded-full transition-colors disabled:pointer-events-none disabled:opacity-50 bg-jungle text-white hover:bg-jungle/90 font-normal text-base px-6 py-2 mt-2 w-fit gap-2"
        href="/offers"
      >
        View our offers
      </a>
    </div>
  </div>
</div>
`;
